src/issue/screens/issue.screen.js

```diff
@@ -160,10 +159,11 @@ class Issue extends Component {
         issueURLParam ? issueCommentsURL : issueParam.comments_url
       ),
     ]).then(() => {
+      const issue = this.props.issue;
+
       if (
-        issueParam &&
         repository.full_name !==
-          issueParam.repository_url.replace(`${apiRoot}/repos/`, '')
```

Comment:
So this was the culprit. It looks like `issueParam`, unlike `props.issue`, is actually obtained from the navigation.params so it's _not_ being updated when those calls within `Promise.all` are fired

Reply:
Suppose we have visited repository A, and try to visit another issue/pr of repository B. `props.repository` will refer to A at first. `issueParam.repository_url` will refer to B.

The basic logic is correct here. But `issueParam.repository_url` does not always have value, which may be caused by `issueParam` itself or `issueParam.repository_url` as my PR description says.

Reply:
@housseindjirdeh Any feedbacks? :)